### PR TITLE
[Term Entry] Add Lua Coroutines .yield()

### LINKED ISSUE #3384: [Term Entry] Lua Coroutines .yield()
### Reason/inspiration (optional)

We would like a new term entry on the `.yield()` method for Lua: [Coroutines](https://www.codecademy.com/resources/docs/lua/coroutines). The entry should go in a new file under `docs/content/lua/concepts/coroutines/terms/yield/yield.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```lua `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by krushnarout**
### Description

Added new term entry on the .yield() method for Lua Coroutines.

Closes #3384 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@krushnarout I've added my initial comments. Please update the entry, and tag me when you've completed the changes. Thanks

File: content/lua/concepts/coroutines/terms/yield/yield.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  print("Hello, Codecademy!")
+  coroutine.yield()
+  print("How are you?")
+end
+
+co = coroutine.create(myCoroutine)
+coroutine.resume(co)
+coroutine.resume(co)
+```
+
```
[the remark this answers is not in the document]
Comment (krushnarout) [reply content/lua/concepts/coroutines/terms/yield/yield.md:43]: @caupolicandiaz I think this should be on 44 line. Can you check again?

[the remark this answers is not in the document]
Comment (krushnarout) [reply content/lua/concepts/coroutines/terms/yield/yield.md:43]: I believe this should be placed on line 44. Can you double-check?

[the remark this answers is not in the document]
Comment (krushnarout) [reply content/lua/concepts/coroutines/terms/yield/yield.md:43]: @caupolicandiaz I believe this should be placed on line 44. Can you double-check?

**Comment by krushnarout** [comment]
@caupolicandiaz I updated the entry with the suggested changes. Please review and let me know if any further changes are needed. Thanks!

**Review by caupolicandiaz — APPROVED** [review]
@krushnarout This is ready for a second review.

**Review by SSwiniarski — APPROVED** [review]
@krushnarout Looks good to me

**Comment by krushnarout** [comment]
@SSwiniarski Thank you

**Pull request merged**